src/NuGet.Core/NuGet.Commands/PackCommandRunner.cs

```diff
@@ -244,26 +245,38 @@ private static void AddDependencyGroups(IList<LibraryDependency> dependencies, N
                     continue;
                 }
 
-                List<string> includes = new List<string>();
-                if (effectiveInclude == LibraryIncludeFlags.All)
+                if (isTargetFrameworks && dependency.LibraryRange.TypeConstraint == LibraryDependencyTarget.Project)
                 {
-                    includes.Add(LibraryIncludeFlags.All.ToString());
+                    continue;
                 }
-                else if ((effectiveInclude & LibraryIncludeFlags.ContentFiles) == LibraryIncludeFlags.ContentFiles)
+                else if (dependency.LibraryRange.TypeConstraint == LibraryDependencyTarget.Reference)
                 {
-                    includes.Add(LibraryIncludeFlags.ContentFiles.ToString());
+                    builder.FrameworkReferences.Add(new FrameworkAssemblyReference(dependency.Name, new NuGetFramework[] { framework }));
+                    addedFrameworkReference = true;
                 }
-
-                List<string> excludes = new List<string>();
-                if ((LibraryIncludeFlagUtils.NoContent & ~effectiveInclude) != LibraryIncludeFlags.None)
+                else
                 {
-                    excludes.AddRange((LibraryIncludeFlagUtils.NoContent & ~effectiveInclude).ToString().Split(new char[] { ',', ' ' }, StringSplitOptions.RemoveEmptyEntries));
-                }
+                    List<string> includes = new List<string>();
+                    if (effectiveInclude == LibraryIncludeFlags.All)
+                    {
+                        includes.Add(LibraryIncludeFlags.All.ToString());
+                    }
+                    else if ((effectiveInclude & LibraryIncludeFlags.ContentFiles) == LibraryIncludeFlags.ContentFiles)
+                    {
+                        includes.Add(LibraryIncludeFlags.ContentFiles.ToString());
+                    }
 
-                packageDependencies.Add(new PackageDependency(dependency.Name, dependency.LibraryRange.VersionRange, includes, excludes));
+                    List<string> excludes = new List<string>();
+                    if ((LibraryIncludeFlagUtils.NoContent & ~effectiveInclude) != LibraryIncludeFlags.None)
+                    {
+                        excludes.AddRange((LibraryIncludeFlagUtils.NoContent & ~effectiveInclude).ToString().Split(new char[] { ',', ' ' }, StringSplitOptions.RemoveEmptyEntries));
+                    }
+
+                    packageDependencies.Add(new PackageDependency(dependency.Name, dependency.LibraryRange.VersionRange, includes, excludes));
+                }
             }
 
-            if (packageDependencies.Any())
+            if (addedFrameworkReference || packageDependencies.Any())
```

Comment:
how do framework references affect this? Are they also part of build.DependencyGroups?